src/main/java/ch/cern/eam/wshub/core/services/contractmanagement/impl/EquipmentReservationAdjustmentServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+    }
+
+    @Override
+    public String deleteEquipmentReservationAdjustment(InforContext context, String number) throws InforException {
+        return tools.performInforOperation(context, inforws::deleteCustomerRentalAdjustmentOp, createDeleteOperation(number)).getResultData().getCUSTOMERRENTALADJUSTMENTID().getCUSTOMERRENTALADJUSTMENTPK();
+    }
+
+    private MP7864_AddCustomerRentalAdjustment_001 createAddOperation(InforContext context, EquipmentReservationAdjustment equipmentReservationAdjustment) throws InforException {
+        MP7864_AddCustomerRentalAdjustment_001 addCustomerRentalAdjustment = new MP7864_AddCustomerRentalAdjustment_001();
+        addCustomerRentalAdjustment.setCustomerRentalAdjustment(tools.getInforFieldTools()
+                .transformWSHubObject(new net.datastream.schemas.mp_entities.customerrentaladjustment_001.CustomerRentalAdjustment(), validateCreateObject(equipmentReservationAdjustment), context));
+        return addCustomerRentalAdjustment;
+    }
+
+    private MP7863_GetCustomerRentalAdjustment_001 createGetOperation(String number) {
+        MP7863_GetCustomerRentalAdjustment_001 getCustomerRentalAdjustment = new MP7863_GetCustomerRentalAdjustment_001();
+        CUSTOMERRENTALADJUSTMENTID_Type idType = new CUSTOMERRENTALADJUSTMENTID_Type();
+        idType.setCUSTOMERRENTALADJUSTMENTPK(number);
+        getCustomerRentalAdjustment.setCUSTOMERRENTALADJUSTMENTID(idType);
+        return getCustomerRentalAdjustment;
+    }
+
+    private MP7865_SyncCustomerRentalAdjustment_001 createUpdateOperation(InforContext context, EquipmentReservationAdjustment equipmentReservationAdjustment) throws InforException {
+        MP7865_SyncCustomerRentalAdjustment_001 syncCustomerRentalAdjustment = new MP7865_SyncCustomerRentalAdjustment_001();
+        syncCustomerRentalAdjustment.setCustomerRentalAdjustment(tools.getInforFieldTools()
+                .transformWSHubObject(new net.datastream.schemas.mp_entities.customerrentaladjustment_001.CustomerRentalAdjustment(), validateUpdateObject(equipmentReservationAdjustment), context));
+        return syncCustomerRentalAdjustment;
+    }
+
+    private MP7866_DeleteCustomerRentalAdjustment_001 createDeleteOperation(String number) {
+        MP7866_DeleteCustomerRentalAdjustment_001 deleteCustomerRentalAdjustment = new MP7866_DeleteCustomerRentalAdjustment_001();
+        CUSTOMERRENTALADJUSTMENTID_Type idType = new CUSTOMERRENTALADJUSTMENTID_Type();
+        idType.setCUSTOMERRENTALADJUSTMENTPK(number);
+        deleteCustomerRentalAdjustment.setCUSTOMERRENTALADJUSTMENTID(idType);
+        return deleteCustomerRentalAdjustment;
+    }
+
+    private EquipmentReservationAdjustment validateCreateObject(EquipmentReservationAdjustment equipmentReservationAdjustment) {
+        validateUpdateObject(equipmentReservationAdjustment);
+        if (isEmpty(equipmentReservationAdjustment.getCode())) {
+            equipmentReservationAdjustment.setCode("1");
```

Comment:
Looking at other examples, if an empty string works here we use that one, otherwise the 0 is the more common 
m.o. 

In addition, if you set it on the EAM object instead of our DTO you don't need to check if it's empty.